uberenv.py

```diff
@@ -770,6 +776,17 @@ def patch(self):
                 print("[copying patched packages from {0}]".format(_src_glob))
                 sexe("cp -Rf {0} {1}".format(_src_glob, dest_spack_pkgs))
 
+        # Update spack's config.yaml if clingo was requested
+        if self.use_clingo:
+            import yaml
+            with open(pjoin(spack_etc_defaults_dir, "config.yaml"), 'r') as conf_file:
+                conf_data = conf_file.read()
+            conf_yaml = yaml.load(conf_data)
+            conf_yaml['config']['concretizer'] = 'clingo'
+            # This will leave out all the comments, is this a problem?
+            with open(pjoin(spack_etc_defaults_dir, "config.yaml"), 'w') as conf_file:
```

Comment:
this is a copy that is placed in the uberenv spack dir right? I think its fine if comments are stripped. 

Q: could you check if the file already has `clingo` specified, and if so skip the injection?

Reply:
I believe you could do
`spack config --scope defaults add config:concretizer:clingo`
and save yourself some trouble here.

Reply:
using spack config command would avoid the pyyaml req for uberenv, so that's a really good adjustment